sentry-core/src/main/java/io/sentry/core/Hub.java

```diff
@@ -484,4 +504,39 @@ public void flush(long timeoutMills) {
     }
     return clone;
   }
+
+  /**
+   * Returns the client bound to this hub
+   *
+   * @return the ISentryClient object
+   */
+  private ISentryClient getSentryClient() {
+    ISentryClient sentryClient = NoOpSentryClient.getInstance();
+    if (!isEnabled()) {
+      options
+          .getLogger()
+          .log(
+              SentryLevel.WARNING,
+              "Instance is disabled and this 'getSentryClient' call is a no-op.");
+    } else {
+      try {
+        StackItem item = stack.peek();
+        if (item != null) {
+          sentryClient = item.client;
+        } else {
+          options.getLogger().log(SentryLevel.FATAL, "Stack peek was null when getSentryClient");
+        }
+      } catch (Exception e) {
+        options
+            .getLogger()
+            .log(SentryLevel.ERROR, "Error while getSentryClient: " + e.getMessage(), e);
+      }
+    }
+    return sentryClient;
+  }
+
+  @Override
+  public <T extends Integration> boolean isIntegrationEnabled(Class<T> integration) {
```

Comment:
This needs to take the actual instance. So when we do `new AnrIntegration` we're matching against this actual object.

If you had 3 instances of the same type in the list, it would return true to all of them and 3 integrations would run.

Reply:
that's true, but on the other hand, if we do by reference, a user could `new UncaughtExceptionHandlerIntegration` and have 2 global hooks, unless I protect those integrations with a static field, what do you think?

Reply:
@HazAT do you have options here?

Reply:
@marandaneto is right, we just check the class not the instance. The reason is that we don't want global stuff to run twice. So we need to make sure that every Client still get the instance of the integration (in case the integration has different options then for another client) but only if `register` was never called for this class before we should do it.

Reply:
The integration instance must see if itself is in the current hub, not its type?

@mitsuhiko last we discussed it was about the instance, so now I'm confused.
Isn't there a difference between _global integrations_ and non global?